TypeSystem/flow-analysis/reachability_A24_t04.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Null-aware method invocation
+/// For now (April, 2025) the flow-analysis feature specification does not
+/// specify the analysis of Null-aware method invocation.
+///
+/// @description Checks reachability after method call. Test return type `Never`
+/// @author sgrekhov@unipro.ru
+
+// SharedOptions=--enable-experiment=sound-flow-analysis
+
+class C {
+  Never m1() => throw "Exception";
+}
+
+test1(C? c) {
+  try {
+    late int i;
+    if (c == null) {  // This doesn't promote `c` to `Null`
+      c?.m1();
+      i = 42;
+    }
+    i; // Possibly assigned
```

Comment:
Perhaps this test is a little bit loose?

Intuitively, `i` can be 'possibly assigned' at `i;` because `c` could be null, and then `c?.m1()` would be a no-op, and `i = 42` would be executed.

We'd be analyzing `c?.m1()` under the assumption that `c` has type `C?`, so it might be a no-op and it might throw (as opposed to: `c` has type `Null` and `c?.m1()` is definitely a no-op). So the `if` statement body will either throw or it will definitely initialize `i`.

Doesn't this mean that we can put `i = 42` into the else branch and achieve a situation where `i` will be definitely assigned after the `if` statement, and we can remove `late`?

Or perhaps this just turns the test into a duplicate of another test?

Reply:
Thank you! No, the change won't make this test a duplicate of another one. Updated. PTAL.